tools/thanosconvert/thanosconvert_test.go

```diff
@@ -0,0 +1,355 @@
+package thanosconvert
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/weaveworks/common/logging"
+
+	"github.com/cortexproject/cortex/pkg/storage/bucket"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+func stringNotEmpty(v string) bool {
+	return v != ""
+}
+
+type fakeBucket map[string]map[string]metadata.Meta
+
+func TestThanosBlockConverter(t *testing.T) {
+	tests := []struct {
+		name       string
+		bucketData fakeBucket
+		assertions func(*testing.T, *bucket.ClientMock, Results, error)
+	}{
+		{
+			name:       "empty bucket is a noop",
+			bucketData: fakeBucket{},
+			assertions: func(t *testing.T, bkt *bucket.ClientMock, results Results, err error) {
+				bkt.AssertNotCalled(t, "Get", mock.Anything, mock.Anything)
+				bkt.AssertNotCalled(t, "Upload", mock.Anything, mock.Anything, mock.Anything)
+				assert.Len(t, results, 0, "expected no users in results")
+			},
+		},
+		{
+			name:       "user with no blocks is a noop",
+			bucketData: fakeBucket{"user1": map[string]metadata.Meta{}},
+			assertions: func(t *testing.T, bkt *bucket.ClientMock, results Results, err error) {
+				bkt.AssertNotCalled(t, "Get", mock.Anything, mock.Anything)
+				bkt.AssertNotCalled(t, "Upload", mock.Anything, mock.Anything, mock.Anything)
+				assert.Contains(t, results, "user1")
+				assert.Len(t, results["user1"].FailedBlocks, 0)
+				assert.Len(t, results["user1"].ConvertedBlocks, 0)
+				assert.Len(t, results["user1"].UnchangedBlocks, 0)
+			},
+		},
+		{
+			name: "bucket fully converted is a noop",
+			bucketData: fakeBucket{
+				"user1": map[string]metadata.Meta{
+					"block1": CortexMeta("user1"),
```

Comment:
Once you start using `ulid.Parse()` these tests will fail because `"block1"` is not a valid block ID. You could just define at the beginning of the test function something like the following, and then use such variables across the test cases:

```
block1 := ulid.MustNew(1, nil)
block2 := ulid.MustNew(2, nil)
```